OD_handler/OD_handler.py

```diff
@@ -0,0 +1,464 @@
+# -*- coding: utf-8 -*-
+"""Main module."""
+
+### Libraries ###
+import pandas as pd
+from datetime import datetime
+import croissance
+from croissance import process_curve
+from croissance.estimation.outliers import remove_outliers
+import re
+import os
+import matplotlib.pyplot as plt 
+import matplotlib
+import numpy as np
+from scipy.optimize import curve_fit
+from croissance.estimation.util import with_overhangs
+from croissance.estimation import regression
+from pandas import Series
+import subprocess
+import sys
+from scipy import interpolate
+import math
+import argparse
+
+__version__ = "0.0.1"
+
+
+
+# ------ MAIN RUNNER ------
+
+
+def argument_parser(argv_list=None):
+	'''Asesses the input arguments and outputs flags to run the different functions according to the user needs.
+	Keyword arguments:
+    argv_list -- List of arguments provided by the user when running the program.
+    Returned variables:
+    flags
+	'''
+
+	#Initialize the argument parser
+	parser = argparse.ArgumentParser()
+
+	#Adding
+	parser.add_argument("-e", "--estimations", help="Get only the estimations for every sample", action = "store_true")
+	parser.add_argument("-f", "--figures", help="Get only the growth curve figures", action = "store_true")
+	parser.add_argument("-s", "--summary", help="Get only the summary of growth rate estimations",action = "store_true")
+	parser.add_argument("-b", "--bioshaker", help="Get one growth rates figures for every individual bioshaker", action = "store_true")
+	parser.add_argument("-i", "--individual", help="Get one growth rates figures for every individual sample", action = "store_true")
+	parser.parse_args()
+	args = parser.parse_args(argv_list[1:])
+
+	#Create flags 
+
+	if args.estimations == False and args.figures == False and args.individual == False and  args.summary == False :
+		
+		flag_all = True
+		flag_est = False
+		flag_sum = False
+		flag_fig = False
+		flag_ind = False
+		flag_bioshaker = False
+
+	elif args.estimations == False and args.figures == False and args.individual == True and  args.summary == False :
+
+		flag_all = True
+		flag_est = False
+		flag_sum = False
+		flag_fig = False
+		flag_ind = True
+		flag_bioshaker = False
+
+	else :
+
+		flag_all = False
+		flag_est = args.estimations
+		flag_sum = args.summary
+		flag_fig = args.figures
+		flag_ind = args.individual
+		flag_bioshaker = args.bioshaker
+
+	return flag_all, flag_est, flag_sum, flag_fig, flag_ind, flag_bioshaker
+
+
+
+# ------ PARSE THE DATA USING THE autoflow_parser LIBRARY ------
+
+
+def parse_data() :
+	'''Calls the autoflow_parser and returns a merged xlsx document with all the OD readings combined'''
+	try :
```

Comment:
Very nice and appropriate use of a try except block.  Try except blocks work great for calling sub processes or trying to read external files that may or may not exist.

I am curious though if it is necessary to call the autoflow_parser as a subprocess or could it be called directly from the library?  There are pros and cons to both approaches and I am curious why you have chosen one over the other.